### PR TITLE
fix(analytics): make ABTestResponse and VariantResponse consistent with Go monorepo

### PR DESCRIPTION
**Pull request by VladislavFitz**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | yes
| New feature?      | no
| BC breaks?        | no     
| Need Doc update   | no

## Describe your change

This PR makes the  `ABTestResponse` and `VariantResponse` structures consistent with the Algolia Go monorepo implementations.

### CODE REVIEW

**Review by rayrutjes — COMMENTED** [review]
I've tried to match the fields here: https://github.com/algolia/go/blob/master/cmd/analytics/api/abtests/response/response.go

Let me know if there was a reason not to do so.

File: algolia/analytics/types_ab_testing.go [hunk 1]
```diff
@@ -37,6 +37,6 @@ func (abTest ABTest) MarshalJSON() ([]byte, error) {
 type Variant struct {
 	Index                  string              `json:"index"`
 	TrafficPercentage      int                 `json:"trafficPercentage"`
```
[the remark this answers is not in the document]
Comment (VladislavFitz) [reply algolia/analytics/types_ab_testing.go:39]: Hey @rayrutjes 
The usage of the signed/size-specific types is not common in the client codebase. We only use `int64` for TaskID and that's why I reduced this information. Do you see any potential drawbacks of lack of precise typing?

**Comment by rayrutjes** [comment]
Hey @VladislavFitz ,

We are currently affected by the lack of `updatedAt` field on the Dynamic Re-Ranking project.

Do you think we could bring this one to life again? Maybe we could focus just on the addition of missing fields, as the change of types to pointers is actually not an easy change to rollout.

Let me know what you think.

**Comment by VladislavFitz** [comment]
@rayrutjes I restored the original types to avoid breaking changes keeping the new values. Let me know if something is missing.

**Review by rayrutjes — COMMENTED** [review]
LGTM except I don't know if the circleci changes belong here, but you are better positioned to answer that.

I'd love an additional review from someone like @marchelbling and/or @JonathanMontane .

**Review by rayrutjes — APPROVED** [review]

**Review by rayrutjes — APPROVED** [review]

**Pull request merged**